README.md

```diff
@@ -173,6 +189,59 @@ D (5079) camera_demo: Done
 @@@@@@@@@        :@@@@@@@@@@.        :@@@@@@@@@@*.     . .@@@@@@@@@@       -@@@@
 ```
 
+## Tweak
+
+Application is configured to run the camera chip with the system clock signal `XCLK` at 10MHz. According to OV7725 specification ver. 1.31 on August 7, 2007, the frequency range of `XCLK` is from 10 to 48MHz. Go ahead and modify this frequency. See [Issue #2](https://github.com/igrr/esp32-cam-demo/issues/2#issuecomment-263345597) for extra guidance.
+
+By changing XCLK you can get your images from camera couple of times faster. For reference see comparison below checked with a scope:
+
+| XCLK frequency | 10MHz | 20MHz | 40MHz |
+| --- | --- | --- | --- |
+| Time to output one pixel / two bytes | 0.2us | 0.1us | 0.05us |
+| Time to output one QVGA line 320 pixels / 640 bytes | 64us | 32us | 16us |
+| Time to output whole QVGA frame of 153,600 bytes | 27.6ms | 13.8ms | 6.9ms |
+
+Note that OV7725 is transmitting two bytes per pixel, so the whole QVGA (320x240 pixels) image takes 153,600 bytes. The [esp32-cam-demo](https://github.com/igrr/esp32-cam-demo) application is currently set up to retrieve images in YUV format. First byte contains Y (luminance) component, the second byte contains UV (chrominance). Application is discarding chrominance byte and saving only the luminance to get gray scale image. As result the image that finally makes to the frame buffer on ESP32 takes 76,800 bytes (half of 153,600 bytes).
+
+The best way to verify timing of signals is with a scope. If you do not have one, then you can still check the millisecond clock printed out on the log.
+
+```
+D (122913) camera: Waiting for VSYNC
+D (122913) camera: Got VSYNC
+D (122913) camera: Waiting for frame
+D (122923) camera: Frame done
+D (122923) camera_demo: Done
+```
+Above example shows [esp32-cam-demo](https://github.com/igrr/esp32-cam-demo) application with `XCLK` running at 40MHz. Every frame is transmitted within 10ms. 
+
+## Troubleshooting
+
+If you have issues to get the live image right, enable test pattern. To do so, change the following define in file `camera.c` from `0` to `1`:
+
+```
+# define ENABLE_TEST_PATTERN 0
+``` 
+
+Camera sensor will then output test pattern instead of live image.
+
+```
+D (51338) camera: Waiting for VSYNC
+D (51358) camera: Got VSYNC
+D (51358) camera: Waiting for frame
+D (51388) camera: Frame done
+D (51388) camera_demo: Done
+ @@@@@@@@@@@@@@@@@@@@@%%%%%%%%%########## +++++++++==========-:::::::::
+  @@@@@@@@@@@@@@@@@@@@@%%%%%%%%%########## +++++++++==========-:::::::::
+   @@@@@@@@@@@@@@@@@@@@%%%%%%%%%##########*+++++++++===========:::::::::.
```

Comment:
This slope, by the way, explains the thing with 2 pixel margin mentioned in one of the other issues.
I wonder if we can improve VSYNC detection (using a GPIO interrupt, maybe?) to avoid having to make such a hack?